File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/webapp/src/ds/CustomerGroupAcctDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.md.bp.ds.model.CustomerGroupAcctDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"custGroupId", type:"int", useNull:true},
		{name:"custGroup", type:"string"},
		{name:"accSchemaId", type:"int", useNull:true},
		{name:"accSchema", type:"string"},
		{name:"salesAccountId", type:"int", useNull:true},
		{name:"salesAccount", type:"string"},
		{name:"prepayAccountId", type:"int", useNull:true},
		{name:"prepayAccount", type:"string"}
	]
});

Ext.define("net.nan21.dnet.module.md.bp.ds.model.CustomerGroupAcctDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"custGroupId", type:"int", useNull:true},
		{name:"custGroupId_From",type:"int", useNull:true},
		{name:"custGroupId_To",type:"int", useNull:true},
		{name:"custGroup", type:"string"},
		{name:"accSchemaId", type:"int", useNull:true},
		{name:"accSchemaId_From",type:"int", useNull:true},
		{name:"accSchemaId_To",type:"int", useNull:true},
		{name:"accSchema", type:"string"},
		{name:"salesAccountId", type:"int", useNull:true},
		{name:"salesAccountId_From",type:"int", useNull:true},
		{name:"salesAccountId_To",type:"int", useNull:true},
		{name:"salesAccount", type:"string"},
		{name:"prepayAccountId", type:"int", useNull:true},
		{name:"prepayAccountId_From",type:"int", useNull:true},
		{name:"prepayAccountId_To",type:"int", useNull:true},
		{name:"prepayAccount", type:"string"}
	] 
});
	
Ext.define("net.nan21.dnet.module.md.bp.ds.param.CustomerGroupAcctDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/bp/ds/model/CustomerGroupAcctDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.md.bp.ds.model;

import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;

import net.nan21.dnet.module.md.bp.domain.entity.CustomerGroupAcct;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = CustomerGroupAcct.class)
public class CustomerGroupAcctDs extends AbstractAuditableDs<CustomerGroupAcct> {

    public static final String fCUSTGROUPID = "custGroupId";
    public static final String fCUSTGROUP = "custGroup";
    public static final String fACCSCHEMAID = "accSchemaId";
    public static final String fACCSCHEMA = "accSchema";
    public static final String fSALESACCOUNTID = "salesAccountId";
    public static final String fSALESACCOUNT = "salesAccount";
    public static final String fPREPAYACCOUNTID = "prepayAccountId";
    public static final String fPREPAYACCOUNT = "prepayAccount";

    @DsField(join = "left", path = "custGroup.id")
    private Long custGroupId;

    @DsField(join = "left", path = "custGroup.code")
    private String custGroup;

    @DsField(join = "left", path = "accSchema.id")
    private Long accSchemaId;

    @DsField(join = "left", path = "accSchema.code")
    private String accSchema;

    @DsField(join = "left", path = "salesAccount.id")
    private Long salesAccountId;

    @DsField(join = "left", path = "salesAccount.code")
    private String salesAccount;

    @DsField(join = "left", path = "prepayAccount.id")
    private Long prepayAccountId;

    @DsField(join = "left", path = "prepayAccount.code")
    private String prepayAccount;

    public CustomerGroupAcctDs() {
        super();
    }

    public CustomerGroupAcctDs(CustomerGroupAcct e) {
        super(e);
    }

    public Long getCustGroupId() {
        return this.custGroupId;
    }

    public void setCustGroupId(Long custGroupId) {
        this.custGroupId = custGroupId;
    }

    public String getCustGroup() {
        return this.custGroup;
    }

    public void setCustGroup(String custGroup) {
        this.custGroup = custGroup;
    }

    public Long getAccSchemaId() {
        return this.accSchemaId;
    }

    public void setAccSchemaId(Long accSchemaId) {
        this.accSchemaId = accSchemaId;
    }

    public String getAccSchema() {
        return this.accSchema;
    }

    public void setAccSchema(String accSchema) {
        this.accSchema = accSchema;
    }

    public Long getSalesAccountId() {
        return this.salesAccountId;
    }

    public void setSalesAccountId(Long salesAccountId) {
        this.salesAccountId = salesAccountId;
    }

    public String getSalesAccount() {
        return this.salesAccount;
    }

    public void setSalesAccount(String salesAccount) {
        this.salesAccount = salesAccount;
    }

    public Long getPrepayAccountId() {
        return this.prepayAccountId;
    }

    public void setPrepayAccountId(Long prepayAccountId) {
        this.prepayAccountId = prepayAccountId;
    }

    public String getPrepayAccount() {
        return this.prepayAccount;
    }

    public void setPrepayAccount(String prepayAccount) {
        this.prepayAccount = prepayAccount;
    }

}
